addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitGarageDoor.java

```diff
@@ -0,0 +1,170 @@
+                doorState = DoorState.STOPPED;
+            }
+        } else {
+            logger.info("Heating cooling target mode not available. Relaying value of OFF to Homekit");
+            doorState = DoorState.STOPPED;
+        }
+        return CompletableFuture.completedFuture(doorState);
+    }
+
+    @Override
+    public CompletableFuture<Boolean> getObstructionDetected() {
+        Item item = getItemRegistry().get(obstructionDetectedItemName);
+        OnOffType state = (OnOffType) item.getStateAs(OnOffType.class);
+        if (state == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return CompletableFuture.completedFuture(state == OnOffType.ON);
+    }
+
+    @Override
+    public CompletableFuture<DoorState> getTargetDoorState() {
+        Item item = getItemRegistry().get(targetDoorStateItemName);
+        OnOffType state = (OnOffType) item.getStateAs(OnOffType.class);
+        if (state == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return CompletableFuture.completedFuture(state == OnOffType.ON ? DoorState.OPEN : DoorState.CLOSED);
+    }
+
+    @Override
+    public CompletableFuture<Void> setTargetDoorState(DoorState targetDoorState) throws Exception {
+        SwitchItem item = getGenericItem(targetDoorStateItemName);
+        item.send(targetDoorState == DoorState.OPEN ? OnOffType.ON : OnOffType.OFF);
+
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @Override
+    public void subscribeCurrentDoorState(HomekitCharacteristicChangeCallback callback) {
+        getUpdater().subscribe(getGenericItem(currentDoorStateItemName), callback);
+    }
+
+    @Override
+    public void subscribeObstructionDetected(HomekitCharacteristicChangeCallback callback) {
+        getUpdater().subscribe(getGenericItem(obstructionDetectedItemName), callback);
+    }
+
+    @Override
+    public void subscribeTargetDoorState(HomekitCharacteristicChangeCallback callback) {
+        getUpdater().subscribe(getGenericItem(targetDoorStateItemName), callback);
+    }
+
+    @Override
+    public void unsubscribeCurrentDoorState() {
+        getUpdater().unsubscribe(getGenericItem(currentDoorStateItemName));
+    }
+
+    @Override
+    public void unsubscribeObstructionDetected() {
+        getUpdater().unsubscribe(getGenericItem(obstructionDetectedItemName));
+    }
+
+    @Override
+    public void unsubscribeTargetDoorState() {
+        getUpdater().unsubscribe(getGenericItem(targetDoorStateItemName));
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T extends GenericItem> T getGenericItem(String name) {
+        Item item = getItemRegistry().get(name);
+        if (item == null) {
+            return null;
+        }
+        if (!(item instanceof GenericItem)) {
+            throw new RuntimeException("Expected GenericItem, found " + item.getClass().getCanonicalName());
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.